### PR TITLE
Forge issues and eID redirect exclusion

### PR DESCRIPTION
**Pull request by lochmueller**
Fix two forge issues and exclude redirect in eID context. Please take a look at it...

### CODE REVIEW

**Comment by lochmueller** [comment]
Any kind of review? Thanks for feedback....

File: class.tx_cooluri.php [hunk 1]
```diff
@@ -434,7 +435,9 @@ public static function goForRedirect($params, $ref)
                     $url = 'http://' . $url[0] . '/' . $url[1];
                 }
 
-                Link_Func::redirect($url, 301);
```
Comment (bednee) [inline class.tx_cooluri.php]: Why is this removed?

File: cooluri/link.Functions.php [hunk 2]
```diff
@@ -259,7 +259,7 @@ public static function pageNotFound($lConf) {
       if (!empty($lConf->cache->pagenotfound)) {
           switch ((string)$lConf->cache->pagenotfound->behavior['type']) {
               case 'message': $res = (string)$lConf->cache->pagenotfound->behavior; break;
-              case 'page':    $res = file_get_contents((string)$lConf->cache->pagenotfound->behavior); break;
+              case 'page':    $res = t3lib_div::getUrl((string)$lConf->cache->pagenotfound->behavior); break;
```
Comment (bednee) [inline cooluri/link.Functions.php]: I don't want dependency on TYPO3 classes in this class. CoolUri core is independent on TYPO3. And this is not compatible with 6.2+

**Comment by bednee** [comment]
You are describing problem with eID, yet your pull request contains other stuff and without explanation why it's done.

File: cooluri/link.Functions.php (hunk 2 above)
Comment (lochmueller) [inline cooluri/link.Functions.php]: Hmm... than you have to write wrapper classes... there is also a mysql_error call in this class that is useless in newer PHP and TYPO3 Versions... main problem is, that many hosters do not allow to fetch external pages via file_get_contents. That is the reason that curl take care of this problem. Are there any plans to build a wrapper for this too?

File: class.tx_cooluri.php (hunk 1 above)
Comment (lochmueller) [inline class.tx_cooluri.php]: My fault... wrong remove.

File: cooluri/link.Functions.php (hunk 2 above)
Comment (bednee) [inline cooluri/link.Functions.php]: Looking at it, the whole code could use some refactoring. The wrapper classes need interface and then be injected into the main class. So, I can accept this at the moment, but make it 6.2+ compatible. Just use \TYPO3\CMS\Core\Utility\GeneralUtility instead on t3lib_div and maybe, to be sure, add at beginning before "class"

if (!class_exists('\TYPO3\CMS\Core\Utility\GeneralUtility')) {
    class_alias('t3lib_div','\TYPO3\CMS\Core\Utility\GeneralUtility');
}

**Comment by lochmueller** [comment]
I cleanup and fix lot of stuff.. use my own fork in feature!...
